node/src/store.rs

```diff
@@ -93,6 +110,19 @@ pub trait Store: Send + Sync + Debug {
     /// This method does not validate or verify that `header` is indeed correct.
     async fn append_single_unchecked(&self, header: ExtendedHeader) -> Result<()>;
 
+    /// Returns height of the lowest header that wasn't sampled yet
+    async fn next_unsampled_height(&self) -> Result<u64>;
+
+    /// Sets sampling result for header, updating last sampled height so that it points
+    /// to the next that wasn't verified
+    /// Returns new highest sampled height or error, if occured
+    async fn mark_header_sampled(&self, height: u64, accepted: bool, cids: Vec<Cid>)
+        -> Result<u64>;
+
+    /// Gets the sampling data for height. `Ok(None)` indicates that the sampling data
+    /// wasn't set in store
+    async fn get_sampling_data_for_height(&self, height: u64) -> Result<Option<SamplingMetadata>>;
```

Comment:
what would you say for `set_sampling_metadata` and `get_sampling_metadata`? I think those are a more common and symmetric